### ISSUE TITLE
Enable ECHconfig by default

### ISSUE THREAD

**Issue by Rathmox** [arkenfox/user.js#1685]
I discovered in [this video](https://youtu.be/9SraN9V23WA?t=541) that it can be interesting to enable ECH.
Is it a good idea ?

```
user_pref("network.dns.echconfig.enabled",true);
user_pref("network.dns.http3_echconfig.enabled",true);
```

**Comment by Rathmox** [comment]
ok, I didn't know about what you said.

**Comment by Thorin-Oakenpants** [comment]
also see #1652

**Comment by Vinfall** [comment]
FYI, Cloudflare started rolling out ECH on Sep 29 according to [their blog](https://blog.cloudflare.com/announcing-encrypted-client-hello/) and they enabled ECH for all free zones already (and it can't be disabled). You can check it at CF dashboard -> <website.domain> -> SSL/TLS -> Edge Certificate -> Encrypted ClientHello (ECH).

Maybe it's still too early to enable it in user.js by default, but it surely has some real world support now.

**Comment by Thorin-Oakenpants** [comment]
it'll happen when it happens ... ooh look -> https://blog.mozilla.org/en/products/firefox/encrypted-hello/
> [ECH is now rolling out to Firefox users worldwide](https://support.mozilla.org/kb/understand-encrypted-client-hello)

Since we aren't part of rollouts (or some of them) we'll probably get it later than sooner by default

**Comment by Thorin-Oakenpants** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1856928 - we get ECH by default in FF120

**Comment by Vinfall** [comment]
Err, I don't know if it would have any impact on the Firefox side (sorry if it does not) so just FYI, Cloudflare disabled ECH globally on Oct 11 in a [community announcement](https://community.cloudflare.com/t/early-hints-and-encrypted-client-hello-ech-are-currently-disabled-globally/567730):

> We have sadly had to disable both of these features globally whilst we address a number of issues with them. These issues are unrelated. We are in the process of adding a label to each of the toggles in dashboard to alert that they are disabled.
> 
> We expect to re-enable Early Hints in the coming weeks, with ECH re-enablement for Free coming later in the year with roll-out to those in the beta in early 2024.

Now the beta is postponed to 2024 and optional opt-in is available _later in the year_.

**Comment by denis-g** [comment]
Why is this disabled?
https://www.cloudflare.com/ssl/encrypted-sni/#results

<img width="1226" alt="image" src="https://github.com/user-attachments/assets/bb130edf-bb93-41c9-b836-b7a4e89815c4">
Tested on v128-132

https://support.mozilla.org/en-US/kb/faq-encrypted-client-hello#w_how-do-i-enable-ech-in-firefox

**Comment by Thorin-Oakenpants** [comment]
ECH has been [enabled in Firefox by default since version 119](https://bugzilla.mozilla.org/show_bug.cgi?id=1856928)
- https://support.mozilla.org/en-US/kb/faq-encrypted-client-hello

But you seem to be asking about ESNI. IANAE on networking

@SkewedZeppelin said this somewhere, maybe he can elaborate
> Firefox only supports native HTTPS query on “Windows 11, Linux, Android 10+”, otherwise the built-in resolver is necessary

I can't even parse what you're saying. You're mad it's disabled and the image proves it? You're mad that you enabled it but the test failed? You're mad we haven't enabled it in arkenfox and FK knows what the image represents? You're not mad but confused - join the crowd?

IANAE but that test site is maybe "a bit rubbish" (from a tor browser dev, not going to go into details). https://tls-ech.dev/ is another site that seemed to suit tor browser purposes better - to be clear it was testing Mullvad Browser, not anything over tor

In any case, like hundreds of previous prefs, I am happy to wait on Mozilla to enable it when ready. Less turmoil and shit in the user.js and less bleeding edge and breakage and issues. If you have a compelling reason why we should, then open a new issue (still not sure what it is you're actually asking)

**Comment by SkewedZeppelin** [comment]
ECH should work out of the box on 129 and higher if you're using a modern system as above

119 enables network.dns.echconfig.enabled: https://bugzilla.mozilla.org/show_bug.cgi?id=1856928
> Encrypted Client Hello (ECH) is now available to Firefox users, delivering a more private browsing experience. ECH extends the encryption used in TLS connections to cover more of the handshake and better protect sensitive fields.

129 enables network.dns.native_https_query: https://www.mozilla.org/en-US/firefox/129.0/releasenotes/
> HTTPS DNS records can now be resolved with the operating system's DNS resolver on specific platforms (Windows 11, Linux, Android 10+). Previously this required [DNS over HTTPS](https://support.mozilla.org/kb/firefox-dns-over-https) to be enabled. This capability allows the use of HTTP/3 without needing to use the [Alt-Svc](https://developer.mozilla.org/docs/Web/HTTP/Headers/Alt-Svc) header, upgrades requests to HTTPS when the DNS record is present, and enables wider use of [ECH](https://support.mozilla.org/kb/faq-encrypted-client-hello).

I have a list of capable sites here: http://divested.dev/misc/ech.txt
